src/App.tsx

```diff
@@ -1,72 +1,70 @@
+    setTitle(event.target.value);
+    applyTitle(event.target.value);
+  }
+
+  function handleInputFieldFocus() {
+    setInputFieldFocus(true);
+  }
+
+  function handleSelectPerson(person: Person) {
+    setSelectedPerson(person);
+    setInputFieldFocus(false);
+  }
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedPerson === null || title !== selectedPerson.name
+            ? 'No selected person'
+            : `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
         </h1>
 
         <div className="dropdown is-active">
           <div className="dropdown-trigger">
             <input
+              value={title}
               type="text"
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              onChange={handleTitle}
+              onClick={handleInputFieldFocus}
             />
           </div>
-
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
-              </div>
-            </div>
-          </div>
-        </div>
-
-        <div
-          className="
-            notification
-            is-danger
-            is-light
-            mt-3
-            is-align-self-flex-start
-          "
-          role="alert"
-          data-cy="no-suggestions-message"
-        >
-          <p className="has-text-danger">No matching suggestions</p>
+          {inputFieldFocus && (
+            <DropdownMenu
+              peopleFromServer={filteredUsers}
+              onSelect={handleSelectPerson}
+            />
+          )}
```

Comment:
When the input field is focused and empty, all people should be shown. Currently, the logic does not handle this case. Consider checking if the `title` is empty and setting `filteredUsers` to `peopleFromServer` in that case.